### PR TITLE
code to check if graph is bipartite or not

### LINKED ISSUE #2064: Bipartite graph check in C++

### PR DESCRIPTION
**Pull request by Akansha2202**
Fixes #2064

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- C++
- 

#### Files Added:
- bipartiteCheck.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Akansha2202** [comment]
please review my pull request #2064

File: bipartiteCheck.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// C++program to check if a connected
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Add this file to a folder.
And fix file name.

File: bipartiteCheck.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// to keep a check whether a node is discovered or not
+unordered_map<int,bool> visited;
+
+// to color the vertices of graph with 2 color
+unordered_map<int,int> color;
+
+// function to store the connected nodes
+void add_edge(vector<int> adj[],int u,int v,bool flag=true)
+{
+	adj[u].push_back(v);
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Use 4 spaces instead of tabs.

File: bipartiteCheck.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+using namespace std;
+
+// to keep a check whether a node is discovered or not
+unordered_map<int,bool> visited;
+
+// to color the vertices of graph with 2 color
+unordered_map<int,int> color;
+
+// function to store the connected nodes
+void add_edge(vector<int> adj[],int u,int v,bool flag=true)
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Add space after `,`, `;`

File: bipartiteCheck.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+{
+	adj[u].push_back(v);
+	if(flag){
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Add space before `{`.

File: bipartiteCheck.cpp [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		adj[v].push_back(u);
+	}
+}
+
+// function to check whether a graph is bipartite or not 
+bool isBipartite(vector<int> adj[],int src)
+{
+	// mark present vertic as visited 
+	visited[src]=true;
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Add space around `=`, `==`.

File: bipartiteCheck.cpp [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	return true;
+}
+
+int main(){
+
+	int n=6;
+	vector<int> adj[n+1];
+
+	// adding edges to the graph 
+	add_edge(adj,1,2);
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Take input from user.

File: bipartiteCheck.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+		}
+	}
+
+	return 0;
+}
+
+
+
+
+
```
Comment (jainaman224) [inline bipartiteCheck.cpp]: Add sample input and output.

**Comment by suchi12j** [comment]
Pls assign this to me

**Comment by somya-kapoor** [comment]
@Akansha2202   Are you working on it?

**Comment by Akansha2202** [comment]
yes I'm
sorry for the delay.

**Comment by somya-kapoor** [comment]
@Akansha2202 Are you working on it?

**Comment by Akansha2202** [comment]
yes, ma'am, I made another pull request earlier but forget to give the issue number.
here is the link
https://github.com/jainaman224/Algo_Ds_Notes/pull/2536
Please review it, ma'am.

**Pull request closed without merging**